src/v0/staff.ts

```diff
@@ -95,9 +103,11 @@ export class Staff {
     })
   }
 
-  create(staffMember: StaffMember): Promise<StaffResponse> {
+  create(staffMember: StaffMember, query?: StaffQuery): Promise<StaffResponse> {
     return new Promise(async (resolve, reject) => {
-      const uri = `${this.options.base}${this.endpoint}/${this.options.user}`
+      const base = this.uriHelper.generateBaseUri()
+      const uri = this.uriHelper.generateUriWithQuery(base, query)
```

Comment:
I just realized we should probably not call a function with arguments of incorrect type, i.e. `generateUriWithQuery` wants a `HandlerQuery`, but you're giving it `StaffQuery` in this case (which also doesn't extend from `HandlerQuery`). I made the same mistake in my implementations as well, so no hard feelings.

I wonder why the typescript compiler is not complaining about this. We should probably refactor this at some point...

Reply:
Ohh, yeah. Totally correct!
I forgot about the required query structure. 
It's really strange that TS didn't complain about it.